lib/synapse/service_watcher/zookeeper.rb

```diff
@@ -123,81 +128,89 @@ def create(path)
+        # we discover service instances. if generator_config_path is present and
+        # the value is "disabled", then skip all zk-based discovery of the
+        # generator config (and use the values from the local config.json
+        # instead).
+        case @discovery.fetch('generator_config_path', nil)
+        when 'disabled'
+          discovery_key = nil
+        when nil
+          discovery_key = 'path'
         else
-          # find the numberic id in the node name; used for leader elections if enabled
-          numeric_id = id.split('_').last
-          numeric_id = NUMBERS_RE =~ numeric_id ? numeric_id.to_i : nil
-
-          log.debug "synapse: discovered backend #{name} at #{host}:#{port} for service #{@name}"
-          new_backends << {
-            'name' => name, 'host' => host, 'port' => port,
-            'id' => numeric_id, 'weight' => weight,
-            'haproxy_server_options' => haproxy_server_options,
-            'labels' => labels
-          }
+          discovery_key = 'generator_config_path'
         end
-      end
 
-      # support for a separate 'generator_config_path' key, for reading the
-      # generator config block, that may be different from the 'path' key where
-      # we discover service instances. if generator_config_path is present and
-      # the value is "disabled", then skip all zk-based discovery of the
-      # generator config (and use the values from the local config.json
-      # instead).
-      case @discovery.fetch('generator_config_path', nil)
-      when 'disabled'
-        discovery_key = nil
-      when nil
-        discovery_key = 'path'
-      else
-        discovery_key = 'generator_config_path'
-      end
-
-      if discovery_key
-        node = @zk.get(@discovery[discovery_key], :watch => true)
-        begin
-          new_config_for_generator = parse_service_config(node.first)
-        rescue StandardError => e
-          log.error "synapse: invalid config data in ZK node at #{@discovery[discovery_key]}: #{e}"
+        if discovery_key
+          node = statsd.time('synapse.watcher.zk.get.elapsed_time', tags: ["zk_cluster:#{@zk_cluster}"]) do
+            @zk.get(@discovery[discovery_key], :watch => true)
+          end
+          begin
+            new_config_for_generator = parse_service_config(node.first)
+          rescue StandardError => e
+            log.error "synapse: invalid config data in ZK node at #{@discovery[discovery_key]}: #{e}"
+            new_config_for_generator = {}
+          end
+        else
           new_config_for_generator = {}
         end
-      else
-        new_config_for_generator = {}
-      end
 
-      set_backends(new_backends, new_config_for_generator)
+        set_backends(new_backends, new_config_for_generator)
+      end
     end
 
     # sets up zookeeper callbacks if the data at the discovery path changes
     def watch
       return if @zk.nil?
       log.debug "synapse: setting watch at #{@discovery['path']}"
```

Comment:
can we add a counter metric here so that we know who (`@name`) are watching on a particular service (@discoery['path'])